libctru/source/system/syscalls.c

```diff
@@ -28,6 +31,64 @@ struct _reent* __SYSCALL(getreent)()
 	return tv->reent;
 }
 
+int __SYSCALL(clock_gettime)(clockid_t clock_id, struct timespec *tp) {
+
+	if (clock_id == CLOCK_REALTIME)
+	{
+		if (tp != NULL)
+		{
+			// Retrieve current time, adjusting epoch from 1900 to 1970
+			s64 ms_since_epoch = osGetTime() - 2208988800000ULL;
+			tp->tv_sec = ms_since_epoch / 1000;
+			tp->tv_nsec = (ms_since_epoch - 1000*tp->tv_sec) * 1000000;
```

Comment:
I used
```cpp
tp->tv_nsec = (ms_since_epoch % 1000) * 1000000;
```
and it ended up with less code, presumably because it can use the remainder output from the previous division operation.

Reply:
Yeah, I should have seen that, I just copied what was done in `gettod_r`. I'll change it in both places.